### PR TITLE
Add ouliner manager, cotext manager and cleanup project files

### LINKED ISSUE #118: Add support for different keywords in css classes
[This](https://github.com/yaptide/ui/blob/7d9ca4d5f0e0cab6546a65e48aff58d1bec0725f/src/ThreeEditor/js/Sidebar.Scene.js#L86) should have all of our new types of objects included in the scene.

### PR DESCRIPTION
**Pull request by Derstilon**
image
image
Closes #118

### CODE REVIEW

File: src/ThreeEditor/js/Config.js [hunk 1]
```diff
@@ -51,10 +51,10 @@ function Config() {
...
 			window.localStorage[name] = JSON.stringify(storage);
 
-			console.log(
-				'[' + /\d\d\:\d\d\:\d\d/.exec(new Date())[0] + ']', // eslint-disable-line
-				'Saved config to LocalStorage.'
-			); 
+			// console.log(
+			// 	'[' + /\d\d\:\d\d\:\d\d/.exec(new Date())[0] + ']', // eslint-disable-line
+			// 	'Saved config to LocalStorage.'
+			// );
```
Comment (Derstilon) [inline src/ThreeEditor/js/Config.js:57]:
```suggestion
			 console.log(
			 	'[' + /\d\d\:\d\d\:\d\d/.exec(new Date())[0] + ']', // eslint-disable-line
			 	'Saved config to LocalStorage.'
			 ); 
```

File: src/ThreeEditor/js/Editor.js [hunk 2]
```diff
@@ -121,18 +129,21 @@ export function Editor(container) {
...
-	this.zonesManager = new CSG.ZoneManager(this); //CSG Manager
+	this.zoneManager = new CSG.ZoneManager(this); //CSG Manager
 	this.detectManager = new DetectManager(this); //Detect Manager
 
 	this.beam = new Beam(this);
 	this.sceneHelpers.add(this.beam);
 
 	this.materialsManager = new MaterialsManager();
 
+
```
Comment (Derstilon) [inline src/ThreeEditor/js/Editor.js:144]:
```suggestion
```

File: src/ThreeEditor/js/sidebar/Sidebar.Scene.js [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+	// 	ignoreObjectSelectedSignal = true;
+
+	// 	editor.selectById(parseInt(outliner.getValue()));
+
+	// 	ignoreObjectSelectedSignal = false;
+	// });
+	// outliner.onDblClick(() => {
+	// 	editor.focusById(parseInt(outliner.getValue()));
+	// });
+	// container.add(outliner);
```
Comment (Derstilon) [inline src/ThreeEditor/js/sidebar/Sidebar.Scene.js:133]:
```suggestion
```

File: src/ThreeEditor/js/sidebar/Sidebar.Scene.js [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+		// options.push(buildOption(worldZone, false));
+
+		// options.push(buildOption(editor.beam, false));
+
+		// outliner.setOptions(options);
+
+		// if (editor.selected !== null) {
+		// 	outliner.setValue(editor.selected.id);
+		// }
+
```
Comment (Derstilon) [inline src/ThreeEditor/js/sidebar/Sidebar.Scene.js:282]:
```suggestion
```

File: src/ThreeEditor/js/sidebar/Sidebar.Scene.js [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+	// 			nextParent = nextParent.parent;
+	// 		}
+
+	// 		if (needsRefresh) refreshUI();
+
+	// 		outliner.setValue('objectSelected', object.id);
+	// 	} else {
+	// 		outliner.setValue(null);
+	// 	}
+	// });
```
Comment (Derstilon) [inline src/ThreeEditor/js/sidebar/Sidebar.Scene.js:363]:
```suggestion
```

File: src/ThreeEditor/js/sidebar/Sidebar.OutlinerManager.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    set id(id) {
+        this.outliner.setId(id);
+    }
+    onChange() {
+        const selectedUuid = this.outliner.getValue();
+        this._selected = this.outliner.selectedValue;
+        this.ignoreObjectSelectedSignal = true;
+        this.editor.selectByUuid(selectedUuid);
+        this.ignoreObjectSelectedSignal = false;
+        console.log(this.editor.selected, selectedUuid);
```
Comment (Derstilon) [inline src/ThreeEditor/js/sidebar/Sidebar.OutlinerManager.js:92]:
```suggestion
```

File: src/ThreeEditor/util/CSG/CSGZone.ts [hunk 7]
```diff
@@ -45,22 +46,19 @@ export class Zone extends THREE.Mesh implements ISimulationObject {
 	worker?: Comlink.Remote<IZoneWorker>;
 	readonly debouncedUpdatePreview = debounce(200, false, () => this.updatePreview());
 
-	private editor: Editor;
+	// private editor: Editor;
```
Comment (Derstilon) [inline src/ThreeEditor/util/CSG/CSGZone.ts:49]:
```suggestion
```

File: src/ThreeEditor/js/Config.js [hunk 8]
```diff
@@ -51,10 +51,10 @@ function Config() {
...
 			window.localStorage[name] = JSON.stringify(storage);
 
-			console.log(
-				'[' + /\d\d\:\d\d\:\d\d/.exec(new Date())[0] + ']', // eslint-disable-line
-				'Saved config to LocalStorage.'
-			); 
+			 console.log(
+			 	'[' + /\d\d\:\d\d\:\d\d/.exec(new Date())[0] + ']', // eslint-disable-line
+			 	'Saved config to LocalStorage.'
+			 );
```
Comment (Derstilon) [inline src/ThreeEditor/js/Config.js:57]:
```suggestion
			console.log(
				'[' + /\d\d\:\d\d\:\d\d/.exec(new Date())[0] + ']', // eslint-disable-line
				'Saved config to LocalStorage.'
			);
```

File: src/ThreeEditor/js/viewport/Viewport.js [hunk 9]
```diff
@@ -506,12 +502,6 @@ export function Viewport(
...
 
-	signals.showZonesChanged.add(showZones => {
-		config.showZones = showZones;
-
-		render();
-	});
-
 	signals.selectModeChanged.add(mode => {
 		//TODO: clicking on zones selects them if zoneSelectionMode is enabled
 	});
```
Comment (Derstilon) [inline src/ThreeEditor/js/viewport/Viewport.js:507]:
```suggestion
```

File: src/ThreeEditor/js/viewport/Viewport.js [hunk 10]
```diff
@@ -26,16 +26,14 @@ export function Viewport(
...
 
 	const config = {
 		showSceneHelpers: true,
-		showZones: true,
-		showDetect: true,
 		selectFigures: true,
 		selectZones: false,
 		selectSections: false,
-		visible: false
+		visible: true
```
Comment (Derstilon) [inline src/ThreeEditor/js/viewport/Viewport.js:36]:
```suggestion
		visible: false
```

File: src/ThreeEditor/js/menubar/Menubar.Add.js [hunk 11]
```diff
@@ -32,11 +32,27 @@ function MenubarAdd(editor) {
...
+
+	//https://stackoverflow.com/questions/37090942/how-to-render-clipped-surfaces-as-solid-objects/37093210#37093210
+	material.onBeforeCompile = function (shader) {
+		shader.fragmentShader = shader.fragmentShader.replace(
+			`gl_FragColor = vec4( outgoingLight, diffuseColor.a );`,
+
+			`gl_FragColor = ( gl_FrontFacing ) ? vec4( outgoingLight, diffuseColor.a ) : vec4( diffuse, opacity );`
+		);
+	};
+
```
Comment (ostatni5) [inline src/ThreeEditor/js/menubar/Menubar.Add.js:55]:
```suggestion
```

**Review by ostatni5 — DISMISSED** [review]
Adress comment
Too many changes in one PR!!!

Overall looks good...
obraz

**Review by ostatni5 — APPROVED** [review]

**Pull request merged**